modules/local/orthofinder_prep.nf

```diff
@@ -0,0 +1,58 @@
+process ORTHOFINDER_PREP {
+    tag "Prepping data for OrthoFinder"
+    label 'process_low'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/orthofinder:2.5.4--hdfd78af_0' :
+        'quay.io/biocontainers/orthofinder:2.5.4--hdfd78af_0' }"
+
+    //publishDir(
+    //    path: "${params.outdir}/orthofinder/",
+    //    mode: 'copy',
+    //    saveAs: { fn -> fn.substring(fn.lastIndexOf('/')+1) },
+    //)
+    
+    input:
+    val fasta_dir
+    val mcl_test
+    
+    output:
+    path "*.dmnd", emit: dmd
+    path "*.fa", emit: fa
+    path "SequenceIDs.txt", emit: seqIDs
+    path "SpeciesIDs.txt", emit: sppIDs
+    path "versions.yml" , emit: versions
+
+    script:
+    def testing_mcl = mcl_test.equals('true') ? "${mcl_test}" : "false"
+    """
+    # Prepare the bespoke orthofinder directory structure and stripped down files
+    # spit the output commands to a tmp file - we don't care about this
+    orthofinder \\
+        -f $fasta_dir \\
+        -t ${task.cpus} \\
+        -op > tmp
+        
+    # Create the publishdir if it's not made yet
+    mkdir -p ../../../${params.outdir}/orthofinder/
+    
+    # Copy all the other orthofinder scraps (species and sequence IDs, etc) to
+    # here - needed for downstream interfacing with orthofinder. 
+    # Specific outdir will depend on whether we're testing mcl or not. 
+    if [ "$testing_mcl" == "true" ]; then
+        outDir="mcl_testing"
+    else
+        outDir="full_analysis"
+    fi
+    
+    mkdir -p ../../../${params.outdir}/orthofinder/\$outDir/
```

Comment:
yeah, it's definitely not ideal

Reply:
Still hacky in a similar way, but I'm wondering if specifying "outdir" as the full path to the directory rather than just the directory name could get rid of the use of relative paths here? Probably best to avoid on principle any solutions that could be described as hacky though, lol.